plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,486 @@
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new IPSLA monitoring policy
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+    sla_type: http
+    http_version: HTTP11
+    http_uri: /test
+    sla_frequency: 120
+    detect_multiplier: 5
+    req_data_size: 64
+    type_of_service: 16
+    operation_timeout: 1000
+    threshold: 1000
+    ipv6_traffic_class: 32
+    state: present
+
+- name: Query an IPSLA monitoring policy with name
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+    state: query
+  register: query_one
+
+- name: Query all IPSLA monitoring policies in the template
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    state: query
+  register: query_all
+
+- name: Delete an IPSLA monitoring policy
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+    state: absent
+
+- name: Query an IPSLA monitoring policy with uuid
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy_uuid: "{{ query_one.current.uuid }}"
+    state: query
+  register: query_one_uuid
+
+- name: Delete an IPSLA monitoring policy with uuid
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy_uuid: "{{ query_one.current.uuid }}"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        ipsla_monitoring_policy=dict(type="str", aliases=["name"]),
+        description=dict(type="str"),
+        ipsla_monitoring_policy_uuid=dict(type="str", aliases=["uuid"]),
+        sla_type=dict(type="str", choices=["http", "tcp", "icmp", "l2ping"]),
```

Comment:
Do we need any specific type / input validation? Like icmp type and http version for instance makes no sense to provide.

Reply:
The API actually accepts values when icmp is used with the optional http type attributes. But I'll go ahead and add a validation.